Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -95,109 +100,151 @@ namespace AzAssetBrowserRequestHandlerPrivate
-                    {
-                        asEditorComponent->SetPrimaryAsset(assetId);
-                    }
-                }
+        if (newEntity == nullptr)
+        {
+            QMessageBox::warning(mainWindow, QObject::tr("Asset Drop Failed"), QStringLiteral("Could not create entity from selected asset(s)."));
+            return;
+        }
+
+        // Deactivate the entity so the properties on the components can be set.
+        newEntity->Deactivate();
+
+        newEntity->SetName(entityName);
+
+
+        // Add all dropped products as components to this entity.
+        AZStd::vector<AZ::EntityId> entityIds = { targetEntityId };
+        EntityCompositionRequests::AddComponentsOutcome addComponentsOutcome = AZ::Failure(AZStd::string());
+        EntityCompositionRequestBus::BroadcastResult(
+            addComponentsOutcome, &EntityCompositionRequests::AddComponentsToEntities, entityIds, componentsToAdd);
 
-                bool isPrefabSystemEnabled = false;
-                AzFramework::ApplicationRequests::Bus::BroadcastResult(
-                    isPrefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+        if (!addComponentsOutcome.IsSuccess())
+        {
+            QMessageBox::warning(mainWindow, QObject::tr("Asset Drop Failed"), QStringLiteral("Could not create components for entity from selected asset(s)."));
+
+            EditorEntityContextRequestBus::Broadcast(&EditorEntityContextRequests::DestroyEditorEntity, targetEntityId);
+            return;
+        }
 
-                if (!isPrefabSystemEnabled)
+        
+        for (const AssetIdAndComponentTypeId& assetAndComponent : assetIdAndComponentTypeId)
+        {
+            AZ::Component* componentAdded = newEntity->FindComponent(assetAndComponent.m_componentTypeId);
+            if (componentAdded)
+            {
+                Components::EditorComponentBase* editorComponent = GetEditorComponent(componentAdded);
+                if (editorComponent)
                 {
-                    // Prepare undo command last so it captures the final state of the entity.
-                    EntityCreateCommand* command = aznew EntityCreateCommand(static_cast<AZ::u64>(newEntity->GetId()));
-                    command->Capture(newEntity);
-                    command->SetParent(undo.GetUndoBatch());
+                    editorComponent->SetPrimaryAsset(assetAndComponent.m_assetId);
                 }
-
-                ToolsApplicationRequests::Bus::Broadcast(&ToolsApplicationRequests::AddDirtyEntity, newEntity->GetId());
-                spawnList.push_back(newEntity->GetId());
             }
         }
+
+        newEntity->Activate();
+
+        bool isPrefabSystemEnabled = false;
+        AzFramework::ApplicationRequests::Bus::BroadcastResult(
+            isPrefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+
+        if (!isPrefabSystemEnabled)
+        {
```

Comment:
just curious why "if prefab system enabled" is being checked here (no real work for this PR)

Reply:
I'm not sure, this code was there before and I tried to preserve it.